packages/fuels-contract/src/contract.rs

```diff
@@ -529,7 +529,7 @@ where
     /// ```ignore
     /// my_contract_instance.my_method(...).set_contracts(&[another_contract_instance]).call()
     /// ```
-    pub fn set_contracts<S: SetableContract>(mut self, contracts: &[&S]) -> Self {
```

Comment:
What was the rationale behind this change?

Reply:
After having a meeting with @supiket, I figured out that you could not make a slice of contract instances that have different types without adding `dyn`. I will add a test to cover this use-case.